tsdb/store.go

```diff
@@ -1124,23 +1124,42 @@ func (s *Store) DeleteMeasurement(database, name string) error {
 	})
 }
 
-// filterShards returns a slice of shards where fn returns true
+// filterShards takes in a list of functions and
+// returns a slice of shards where fn returns true
 // for the shard. If the provided predicate is nil then all shards are returned.
 // filterShards should be called under a lock.
-func (s *Store) filterShards(fn func(sh *Shard) bool) []*Shard {
+func (s *Store) filterShards(fns []func(sh *Shard) bool) []*Shard {
 	var shards []*Shard
-	if fn == nil {
+
+	// Early return if nil or 0 len
+	if len(fns) == 0 || fns == nil {
```

Comment:
`len(fns) == 0` is sufficient, you don't need the separate nil check. You can see this in action at https://go.dev/play/p/XWF--s4NyNn

Also, if you did _need_ a nil check, you would need to put it first to avoid dereferencing a nil pointer.